src/manifestLayout.js

```diff
@@ -267,72 +267,86 @@ var manifestLayout = function(options) {
+  /**
+  * Determines a facing page type
+  * @returns {String}
+  */
+  function facingPageType(index) {
+    if (index === 0) {
+      return 'firstPage';
+    } else if ((index % 2) === 0) {
+      return 'rightPage';
+    } else {
+      return 'leftPage';
+    }
+  }
 
-    lines.currentLine = 0;
+  var Lines = function(lineWidth, frames){
+    this.currentLine = 0;
+    this.lineWidth = lineWidth;
+    this.frames = frames;
+  };
+
+  Lines.prototype = {
+    _getFacingFrame: function(position) {
+      return this.frames.filter(function(page) {
+        var value;
+        switch (facingPageType(position)) {
+          case 'rightPage':
+          value = -1;
+          break;
+          case 'leftPage':
+          value = 1;
+          break;
+        }
+        return page.canvas.sequencePosition + value === position;
+      })[0];
+    },
 
-    lines.addLine = function() {
-      var line = lines[lines.currentLine] = [];
-      line.remaining = lineWidth;
+    addLine: function() {
+      var line = this[this.currentLine] = [];
+      line.remaining = this.lineWidth;
 
       return line;
-    };
+    },
 
     /**
      * @param frame
      * @returns {Array} [frame x position, line frame is on]
      */
-    lines.addItem = function(frame) {
+    addItem: function(frame) {
       var line = this[this.currentLine],
           lineItemWidth,
           x;
 
+      lineItemWidth = frame.width;
       if (viewingMode === 'paged') {
-        var position = frame.canvas.sequencePosition;
-        // Return the facingFrame, based on the facing page type
-        var facingFrame = frames.filter(function(page) {
-          var value;
-          switch (facingPageType(position)) {
-            case 'rightPage':
-            value = -1;
-            break;
-            case 'leftPage':
-            value = 1;
-            break;
-          }
-          return page.canvas.sequencePosition + value === position;
-        })[0];
-
-        if (facingFrame) {
-          lineItemWidth = frame.width + facingFrame.width;
-        } else {
-          lineItemWidth = frame.width;
+        var facingFrame = this._getFacingFrame(frame.canvas.sequencePosition);
+        if(facingFrame){
+          lineItemWidth += facingFrame.width;
         }
-      } else {
-        lineItemWidth = frame.width;
       }
 
       if (!line) { line = this.addLine(); }
 
       if (line.remaining >= lineItemWidth) {
-        x = lineWidth - line.remaining;
+        x = this.lineWidth - line.remaining;
         if (viewingDirection === 'right-to-left') {
           x = line.remaining - frame.x;
         }
         line.remaining -= frame.width;
-        return [x, lines.currentLine];
-      }
-      if (line.remaining >= frame.width && facingPageType(frame.canvas.sequencePosition) === 'rightPage') {
-        x = lineWidth - line.remaining;
-        return [x, lines.currentLine];
+        return [x, this.currentLine];
       }
       this.currentLine += 1;
```

Comment:
Indeed. That won't work with the addLine call on line 330, though, unless we init currentLine to -1 at the beginning. I'll keep it in mind for future refactoring.